src/unity/Assets/Crest/Scripts/Shapes/ShapeGerstnerBatched.cs

```diff
@@ -348,6 +364,92 @@ public Vector3 GetDisplacement(ref Vector3 worldPos, float toff)
             return result;
         }
 
+        public bool m_Approx = true;
+
+        public float SinCos(float x,out float cos)
+        {
+//            if (!m_Approx)
+            {
+                cos = Mathf.Cos(x);
+                return Mathf.Sin(x);
+            }
+//            cos = Mathf.Cos(x);
+//            return Mathf.Sqrt(1 - cos * cos);
+
+
+            /*
+                        float xf = Mathf.Floor(x / (Mathf.PI * 2));
+                        x -= xf * Mathf.PI * 2;
+                        const float B = 4 / Mathf.PI;
+                        const float C = -4 / (Mathf.PI * Mathf.PI);
+
+                        float Sin= -(B * x + C * x * ((x < 0) ? -x : x));
+
+                        cos = Mathf.Sqrt(1 - Sin * Sin);
+
+                        return Sin;
+            */
+        }
+
+        Dictionary<uint,Vector3> m_Recent=new Dictionary<uint, Vector3>();
+
+        uint CalcHash(Vector3 _wp)
+        {
+            _wp *= 10;
+            return (uint)((_wp.x+32000)+(_wp.z+32000)*65000);
+        }
+        public Vector3 GetDisplacementFast(ref Vector3 worldPos, float toff)
```

Comment:
good stuff. im pondering a bit if the cache could be implemented outside this class, to separate the code out and have teh cache as a layer that sits above. it may also help prevent code duplication, although i see you've broken it out because you're using the most significant wavelengths only, and you've optimised the code a bit. but this can all come later, happy with what you've implemented, with minor feedback - do you mind making whitespace consistent and remove any commented code.